Makefile

```
@@ -3,21 +3,56 @@ SHELL := /bin/bash # Use bash syntax
 # Set up variables
 GO111MODULE=on
 
-# Build ldflags
-VERSION ?= "v0.0.0"
-GITCOMMIT=$(shell git rev-parse HEAD)
-BUILDDATE=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
-GO_LDFLAGS=-ldflags "-X main.version=$(VERSION) \
-			-X main.buildHash=$(GITCOMMIT) \
-			-X main.buildDate=$(BUILDDATE)"
+GO_CMD_FLAGS=-tags codegen
+AWS_SERVICE=$(shell echo $(SERVICE))
+SERVICE_MODEL_NAME=$(shell echo $(MODEL_NAME))
+ifeq ($(SERVICE_MODEL_NAME),)
+  SERVICE_MODEL_NAME:=""
+endif
+ROOT_DIR=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+CONTROLLER_BOOTSTRAP=./bin/controller-bootstrap
+CONTROLLER_DIR=${ROOT_DIR}/../${AWS_SERVICE}-controller
+CODE_GEN_DIR=${ROOT_DIR}/../code-generator
+AWS_SDK_GO_VERSION=$(shell curl -H "Accept: application/vnd.github.v3+json" \
+                                 https://api.github.com/repos/aws/aws-sdk-go/releases/latest | jq -r '.tag_name')
+ACK_RUNTIME_VERSION=$(shell curl -H "Accept: application/vnd.github.v3+json" \
+								 https://api.github.com/repos/aws-controllers-k8s/runtime/releases/latest | jq -r '.tag_name')
+.DEFAULT_GOAL=run
+EXISTING_CONTROLLER="true"
+DRY_RUN="false"
 
-.PHONY: all test
+.PHONY: build, generate, update, init, run, clean
 
-all: test
+build:
+	@go build ${GO_CMD_FLAGS} -o ${CONTROLLER_BOOTSTRAP} ./cmd/controller-bootstrap/*.go
```

nit: Point to the specific entrypoint, just in case we add additional entrypoints later
```suggestion
    @go build ${GO_CMD_FLAGS} -o ${CONTROLLER_BOOTSTRAP} ./cmd/controller-bootstrap/main.go
```